src/operators.jl

```diff
@@ -7,6 +7,41 @@ using SparseArrays
 
 function default_dec_cu_matrix_generate() end;
 
+export cache_vcycle_lap, run_multigrid_vcycles, vcycle_lap_op
+
+# TODO: Move vcycle code into CombSpaces
+function cache_vcycle_lap(fs::AbstractVector{<:GeometricMap})
+  sses = map(fs) do f dom(f).delta_set end
+  push!(sses,codom(last(fs)).delta_set) # Get original mesh
+
+  sds = map(sses) do s dualize(s, Circumcenter()) end # TODO: Add option
+
+  Ls = map(sds) do sd ∇²(0,sd) end
+  rs = as_matrix.(fs)./4.0 #4 is the biggest row sum that occurs for triforce, this is not clearly the correct scaling
+  ps = transpose.(as_matrix.(fs))
+  (Ls, rs, ps)
+end
+
+function run_multigrid_vcycles(b,As,rs,ps,steps,cycles=1,alg=cg)
+  multigrid_vcycles(zeros(length(b)),b,As,rs,ps,steps,cycles,alg)
+end
+
+function vcycle_lap_op(fs::AbstractVector{<:GeometricMap})
+  vcyc_cache = cache_vcycle_lap(fs)
+  x -> run_multigrid_vcycles(x, vcyc_cache..., 3, 10, cg)
+end
+
+dom2dual(f::GeometricMap) = dualize(dom(f).delta_set)
+codom2dual(f::GeometricMap) = dualize(codom(f).delta_set)
+
+function default_dec_matrix_generate(fs::AbstractVector{<:GeometricMap}, my_symbol::Symbol, hodge::DiscreteHodge)
+  op = @match my_symbol begin
+    :Δ₀⁻¹ => vcycle_lap_op(fs)
```

Comment:
We will have to consider which Laplacian solve should be the default. Hopefully this could be controlled by the keyword argument `multigrid` from `gensim`, or from dispatch.

Reply:
If you get a geometric map, then it should use multigrid, if not factorize.